components/gitpod-protocol/src/context-url.ts

```diff
@@ -0,0 +1,50 @@
+/**
+ * Copyright (c) 2020 Gitpod GmbH. All rights reserved.
+ * Licensed under the GNU Affero General Public License (AGPL).
+ * See License-AGPL.txt in the project root for license information.
+ */
+
+export namespace ContextURL {
+  export const INCREMENTAL_PREBUILD_PREFIX = "incremental-prebuild";
+  export const PREBUILD_PREFIX = "prebuild";
```

Comment:
I'm not big on this because it centralises the otherwise server local prefix concern in gitpod-protocol.
Can't we add the protocol after the prefix parsing has happened?

Reply:
> Can't we add the protocol after the prefix parsing has happened?

I'm not sure I understand.

We can indeed move this to `server`. I'm just wondering how the frontend is doing URL parsing... I bet if we dig a bit we'll find similar code.

Reply:
I too think this should not be centralized. We should rather test the individual prefix-context-parsers.

Reply:
> We should rather test the individual prefix-context-parsers.

That's also possible. But what about code like [this](https://github.com/gitpod-io/gitpod/blob/1c6895539f5153f1e90af6e7a034665197d8fe8a/components/dashboard/src/start/StartWorkspace.tsx#L152-L153), then?

> I too think this should not be centralized

IMO having a single piece of code returning a `ContextURL` of sorts would be more suitable. The independence of the multiple index parsers feels way too artificial and brittle.

Reply:
That's tricky.

Code places like 

https://github.com/gitpod-io/gitpod/blob/5fcaa6d52b945d0634360f877e8511c67479b38e/components/dashboard/src/start/StartWorkspace.tsx#L152-L153

are pretty awful. There are already prefixes missing there. How to know that you have to adjust this place, when you add a new `IPrefixContextParser`?

On the other hand, having the prefix centralized here counteracts the dependency injection style. Would be great if we could collect the prefixes dynamically somehow.

But all in all, I find this the lesser evil than that prefixes are hard-coded somewhere across the code.

@geropl Is there any good reason that you didn't replace the prefix strings in `StartWorkspace.tsx` with the variables?

Reply:
> @geropl Is there any good reason that you didn't replace the prefix strings in StartWorkspace.tsx with the variables?

I just pushed the commit that uses function above in frontend and backend.